File: sdk/src/main/java/io/horizen/account/api/rpc/types/TxPoolContent.java
package io.horizen.account.api.rpc.types;

import io.horizen.evm.Address;

import java.math.BigInteger;
import java.util.Map;

public class TxPoolContent {
    public final Map<Address, Map<BigInteger, TxPoolTransaction>> pending;
    public final Map<Address, Map<BigInteger, TxPoolTransaction>> queued;

    public TxPoolContent(
        Map<Address, Map<BigInteger, TxPoolTransaction>> pending,
        Map<Address, Map<BigInteger, TxPoolTransaction>> queued
    ) {
        this.pending = pending;
        this.queued = queued;
    }
}

File: sdk/src/main/scala/io/horizen/account/api/rpc/types/TxPoolContent.scala
package io.horizen.account.api.rpc.types

import com.fasterxml.jackson.annotation.JsonView
import io.horizen.evm.Address
import io.horizen.json.Views

import java.math.BigInteger
import scala.collection.concurrent.TrieMap
import scala.collection.mutable

@JsonView(Array(classOf[Views.Default]))
class TxPoolContent(
    val pending: TrieMap[Address, mutable.SortedMap[BigInteger, EthereumTransactionView]],
    val queued: TrieMap[Address, mutable.SortedMap[BigInteger, EthereumTransactionView]]) { }
